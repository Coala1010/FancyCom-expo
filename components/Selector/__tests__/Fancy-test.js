// __tests__/Intro-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { Selector } from '../';
import { MaterialIcons } from '@expo/vector-icons';

const selectorArray = [
  {
    id: 'none',
    icon: <MaterialIcons name='block' size={30} color='#9D9D9D' />,
    label: 'NONE',
  },
  {
    id: 'border',
    icon: <MaterialIcons name='border-outer' size={30} color='#9D9D9D' />,
    label: 'SOLID',
  },
  {
    id: 'dashed',
    icon: <MaterialIcons name='border-clear' size={30} color='#9D9D9D' />,
    label: 'DASHED',
  },
];

describe('Some component', () => {
  it('Should render the component in the dom', () => {
    const tree = renderer.create(
      <Selector
        items={selectorArray}
        onChange={() => console.log('onChange clicked')} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});