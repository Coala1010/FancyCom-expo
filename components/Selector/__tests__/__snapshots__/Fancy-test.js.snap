// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#F4F4F4",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              NONE
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              SOLID
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              DASHED
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`Some component Should redner the component in the dom 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#F4F4F4",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              NONE
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              SOLID
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              DASHED
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`Some component Should render the component in the dom 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#F4F4F4",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              NONE
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              SOLID
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              DASHED
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`Some component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#F4F4F4",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              NONE
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              SOLID
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              DASHED
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#F4F4F4",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              NONE
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              SOLID
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "opacity": 1,
              "paddingHorizontal": 5,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#DCDCDC",
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 60,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              DASHED
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
